### PR TITLE
List of spaces

### PR DESCRIPTION
**Pull request by PiperS52**
Written feature and unit tests for showing a list of spaces on /spaces - rspec passing

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -2,9 +2,11 @@
 require "sinatra/base"
 require "./database_connection_setup"
 require 'bcrypt'
+require_relative './lib/space'
```
Comment (chriswhitehouse) [inline app.rb:5]: Probably better in config.run

File: app.rb [hunk 2]
```diff
@@ -14,6 +16,7 @@ class MakersBnB < Sinatra::Base
   end
 
   get "/spaces" do
+    @spaces = Space.all
```
Comment (chriswhitehouse) [inline app.rb:19]: Cool

File: lib/space.rb [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+    Space.new(
+      id: result[0]['id'],
+      name: result[0]['name'],
+      description: result[0]['description'],
+      date_available_from: result[0]['date_available_from'],
+      date_available_to: result[0]['date_available_to'],
+      price: result[0]['price'])
   end
+  
   attr_reader :id, :name, :description, :price, :date_available_from, :date_available_to
```
Comment (chriswhitehouse) [inline lib/space.rb]: sorry, attr readers are supposed to go at the top

File: spec/features/list_of_spaces_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      description: "Another description",
+      date_available_from: "2021-01-11",
+      date_available_to: "2021-02-12",
+      price: "50")
+
+    visit('/spaces')
+    expect(page).to have_selector("input", :class => "list_a_space")
+    expect(page).not_to have_content('sinatra')
+    expect(page).to have_content('Book A Space')
+    # expect(page).to have_content("Test name") ``
```
Comment (chriswhitehouse) [inline spec/features/list_of_spaces_spec.rb]: Reinstate these expect methods?

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -42,6 +42,10 @@
 SimpleCov.start
 
 RSpec.configure do |config|
+  config.before(:each) do
```
Comment (chriswhitehouse) [inline spec/spec_helper.rb:45]: Nice, thanks for spotting this.

File: spec/units/space_spec.rb [hunk 6]
```diff
@@ -31,4 +31,38 @@
     end
   end
 
+  describe '.all' do
```
Comment (chriswhitehouse) [inline spec/units/space_spec.rb:34]: Nice, looks good.

File: views/spaces/index.erb [hunk 7]
```diff
@@ -1 +1,24 @@
...
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Book A Space</title>
+</head>
+<body>
+<h1>Book A Space</h1>
+<form action="/spaces/new" method="get">
+  <input type="submit" name="list_of_spaces" class="list_a_space" value="List a Space"/>
```
Comment (chriswhitehouse) [inline views/spaces/index.erb]: Note name is plural, class is singular, is this ok?

File: views/spaces/index.erb [hunk 8]
```diff
@@ -1 +1,24 @@
...
+</form>
+<div>
+  <ul class="indiv_space">
```
Comment (chriswhitehouse) [inline views/spaces/index.erb:14]: is the class still a list of spaces?

**Review by chriswhitehouse — COMMENTED** [review]
Thanks Simon for working this through.

File: app.rb (hunk 1 above)
Comment (PiperS52) [reply app.rb:5]: Removed

File: lib/space.rb (hunk 3 above)
Comment (PiperS52) [reply lib/space.rb]: Changed

File: spec/features/list_of_spaces_spec.rb (hunk 4 above)
Comment (PiperS52) [reply spec/features/list_of_spaces_spec.rb]: Done

File: views/spaces/index.erb (hunk 7 above)
Comment (PiperS52) [reply views/spaces/index.erb]: Changed to "list of space"

File: views/spaces/index.erb (hunk 8 above)
Comment (PiperS52) [reply views/spaces/index.erb:14]: No, as this is the class for the 'list a space' button

**Review by chriswhitehouse — APPROVED** [review]

**Comment by chriswhitehouse** [comment]
[As a user So i can choose a space I would like to see a list of spaces](https://trello.com/c/y0hnKAFZ/3-as-a-user-so-i-can-choose-a-space-i-would-like-to-see-a-list-of-spaces)

**Pull request merged**